packages/block/src/header.ts

```diff
@@ -249,7 +249,7 @@ export class BlockHeader {
         }
       }
     } else {
-      if (baseFeePerGas) {
+      if (typeof baseFeePerGas === 'bigint') {
```

Comment:
I think we should just check for existence not type, as `baseFeePerGas` can be any `BigIntLike` value.
```suggestion
      if (baseFeePerGas !== undefined) {
```